### PR TITLE
Docker tutorial

### LINKED ISSUE #11705: Rewrite and adapt ASP.NET Core Docker article from .NET Core repo
There is an article in the .NET Core repo that talks about creating an ASP.NET Core Docker image and container that is more suited to live in this doc set. 

https://github.com/dotnet/docs/blob/ea80a5bdcfc37d09204643aa50a7bae52410261f/docs/core/docker/building-net-docker-images.md

I'm going to delete that article as I publish updates to the Docker area of .NET Core. Please let me know when you get this article rewritten and published and I will link back and provide a redirect from the old page to the new one.

@scottaddie

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #11705
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/docker/building-net-docker-images?view=aspnetcore-1.0&branch=pr-en-us-11929)

### CODE REVIEW

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+> * Learn about Microsoft .NET Core Docker images 
+> * Download an ASP.NET Core sample app
+> * Run the sample app locally
+> * Run the sample app in Linux containers
+> * Run the sample app in Windows containers
+> * Build and deploy manually
+
+## ASP.NET Core Docker images
+
+For this tutorial, you download an ASP.NET Core sample application and run it in Docker containers. The sample works with both Linux and Windows containers.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: For this tutorial, you download an ASP.NET Core sample app and run it in Docker containers. The sample works with both Linux and Windows containers.

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+     * [CentOS](https://docs.docker.com/install/linux/docker-ce/centos/)
+     * [Debian](https://docs.docker.com/install/linux/docker-ce/debian/)
+     * [Fedora](https://docs.docker.com/install/linux/docker-ce/fedora/)
+     * [Ubuntu](https://docs.docker.com/install/linux/docker-ce/ubuntu/)
+  * [macOS](https://docs.docker.com/docker-for-mac/install/)
+  * [Windows](https://docs.docker.com/docker-for-windows/install/)
+
+* [git](https://git-scm.com/download)
+
+## Download the sample application
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: ## Download the sample app

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+* Navigate to the project folder at *dotnet-docker/samples/aspnetapp/aspnetapp*.
+
+* Run the following command to build and run the app locally:
+
+  ```console
+  dotnet run
+  ```
+
+* After the application starts, visit `http://localhost:5000` in a browser to verify that the app is working.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: Navigate to `http://localhost:5000` to test the app.

Comment (Rick-Anderson) [reply aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: You can toss in browser if you want, but I don't think it's needed.

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+* Press Ctrl+C at the command prompt to stop the app.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: at the command prompt or in the command prompt?

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+  * Look for the Dockerfile in the current folder (the period at the end).
+
+  The run command arguments:
+  * Allocate a pseudo-TTY and keep it open even if not attached. (Same effect as `--interactive --tty`.)
+  * Automatically remove the container when it exits.
+  * Map port 5000 on the local machine to port 80 in the container.
+  * Name the container aspnetcore_sample.
+  * Specify the aspnetapp image.
+
+* After the application starts, see the home page at `http://localhost:5000` in your web browser.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: I'd make this similar to the previous.
Why are these using HTTP and not HTTPS? Must be a very old tutorial that hasn't been updated.

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    Ethernet adapter Ethernet:
+
+       Connection-specific DNS Suffix  . : contoso.com
+       Link-local IPv6 Address . . . . . : fe80::1967:6598:124:cfa3%4
+       IPv4 Address. . . . . . . . . . . : 172.29.245.43
+       Subnet Mask . . . . . . . . . . . : 255.255.240.0
+       Default Gateway . . . . . . . . . : 172.29.240.1
+    ```
+
+* Copy the container IPv4 address (for example, 172.29.245.43) and paste into the browser address bar to see the sample app's home page.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: Copy the container IPv4 address (for example, 172.29.245.43) and paste into the browser address bar to test the app.

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md (hunk 4 above)
Comment (tdykstra) [reply aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: at the command prompt or in the command prompt window.

File: aspnetcore/host-and-deploy/docker/building-net-docker-images.md (hunk 5 above)
Comment (tdykstra) [reply aspnetcore/host-and-deploy/docker/building-net-docker-images.md]: HTTP comes from the original tutorial.  I'll make a separate issue to convert to HTTPS.

**Pull request merged**